src/tools/clippy/tests/ui/checked_unwrap/simple_conditionals.rs

```diff
@@ -173,8 +173,10 @@ fn issue11371() {
     static mut X: Option<i32> = Some(123);
     unsafe {
         if X.is_some() {
+            //~^ ERROR: creating a shared reference to mutable static is discouraged
             X = None;
             X.unwrap();
+            //~^ ERROR: creating a shared reference to mutable static is discouraged
```

Comment:
Why is this the only test case (in Clippy)` that didn't get the allow attribute, but rather an updated stderr file?

Also, the two cases in this file only create temporary references that aren't bound to anything. The `unwrap` even takes `X` by move. Is this lint too aggressive here or am I missing something?

Reply:
I think `is_some()` should emit the error as it takes &self, but `unwrap()` should not, as it takes self. So, that's why I asked a question in Zulip: https://rust-lang.zulipchat.com/#narrow/stream/182449-t-compiler.2Fhelp/topic/get.20fnunction.20signature.20of.20method.20call/near/438255044